fqd_dss.c

```diff
@@ -166,11 +172,21 @@ fqd_queue_message_process(remote_data_client *me, fq_msg *msg)
     
   ck_pr_inc_32(&work_queue_backlogs[tindex]);
   work_queue = &work_queues[tindex];
+  ck_fifo_spsc_enqueue_lock(work_queue);
   entry = ck_fifo_spsc_recycle(work_queue);
   if (entry == NULL) {
-    entry = malloc(sizeof(ck_fifo_spsc_entry_t));
+    ck_fifo_spsc_enqueue_unlock(work_queue);
```

Comment:
This section is correct. Not sure we need to unlock to `malloc` but it is more correct in case the allocation takes forever.